ArdusatSDK.cpp

```diff
@@ -453,266 +257,1212 @@ const char * valuesToCSV(const char *sensorName, unsigned long timestamp, int nu
-
-  dtostrf(input.x, 2, 3, _getOutBuf() + _output_buf_len);
-  _output_buf_len = strlen(_getOutBuf());
-  _getOutBuf()[_output_buf_len++] = CSV_SEPARATOR;
-
-  dtostrf(input.y, 2, 3, _getOutBuf() + _output_buf_len);
-  _output_buf_len = strlen(_getOutBuf());
-  _getOutBuf()[_output_buf_len++] = CSV_SEPARATOR;
-
-  dtostrf(input.z, 2, 3, _getOutBuf() + _output_buf_len);
-  _output_buf_len = strlen(_getOutBuf());
-
-  _add_checksum_to_csv_buffer(sensorName, 3, input.x, input.y, input.z);
-  _getOutBuf()[_output_buf_len++] = '\n';
-
+/**
+ * Creates a JSON string with the appropriate values
+ *
+ * @param sensor_name the label to be given to some data
+ * @param unit the unit of measurement used
+ * @param value the sensor value
+ *
+ * @return the output buffer
+ */
+const char * valueToJSON(const char *sensorName, unsigned char unit, float value) {
+  _resetOutBuf();
+  _writeJSONValue(_getOutBuf(), sensorName, unit_to_str(unit), value);
   return _getOutBuf();
 }
 
-_DEFAULT_SENSOR_NAME_TO_CSV_FXN(temperature)
-const char * temperatureToCSV(const char *sensorName, temperature_t & input) {
-  _headerToCSV(&(input.header), sensorName);// warning : resets the internal buffer
-
-  dtostrf(input.t, 2, 3, _getOutBuf() + _output_buf_len);
-  _output_buf_len = strlen(_getOutBuf());
-
-  _add_checksum_to_csv_buffer(sensorName, 1, input.t);
-  _getOutBuf()[_output_buf_len++] = '\n';
 
-  return _getOutBuf();
+/**************************************************************************//**
+ * @brief   Constructs Acceleration sensor object
+ * @ingroup acceleration
+ *
+ * Example Usage:
+ * @code
+ *     Acceleration accel = Acceleration(); // Instantiate sensor object
+ *     accel.begin();                       // Initialize sensor
+ *     Serial.println(accel.readToJSON());  // Read and print values in JSON
+ * @endcode
+ *****************************************************************************/
+Acceleration::Acceleration(void) {
+  Acceleration::sensorId = SENSORID_ADAFRUIT9DOFIMU;
+  Acceleration::gGain = LSM303_ACCEL_GAIN8G;
+  Acceleration::header.unit = DATA_UNIT_METER_PER_SECONDSQUARED;
+  Acceleration::initialized = false;
 }
 
-_DEFAULT_SENSOR_NAME_TO_CSV_FXN(gyro)
-const char * gyroToCSV(const char *sensorName, gyro_t & input) {
-  _headerToCSV(&(input.header), sensorName);// warning : resets the internal buffer
-
-  dtostrf(input.x, 2, 3, _getOutBuf() + _output_buf_len);
-  _output_buf_len = strlen(_getOutBuf());
-  _getOutBuf()[_output_buf_len++] = CSV_SEPARATOR;
+Acceleration::~Acceleration() {
+}
 
-  dtostrf(input.y, 2, 3, _getOutBuf() + _output_buf_len);
-  _output_buf_len = strlen(_getOutBuf());
-  _getOutBuf()[_output_buf_len++] = CSV_SEPARATOR;
+/**
+ * @brief   Initializes the sensor with basic configurations
+ * @ingroup acceleration
+ *
+ * @retval true  Successfully initialized
+ * @retval false Failed to initialize
+ */
+boolean Acceleration::begin(void) {
+  return Acceleration::begin(Acceleration::gGain);
```

Comment:
In general, use `this` rather than `Acceleration::` to reference member variables and references. You _can_ use this syntax when you're positive that you want to only call a specific version of a (potentially overloaded) method, but that's not the usual case. (e.g. `this->begin(...)` in this case)